sample-apps/python-sample-app/app.py

```diff
@@ -0,0 +1,157 @@
+from tkinter import N
+from config import *
+
+import random_metrics
+import random
+import requests
+import boto3
+
+from opentelemetry.propagators.aws.aws_xray_propagator import (
+    TRACE_ID_DELIMITER,
+    TRACE_ID_FIRST_PART_LENGTH,
+    TRACE_ID_VERSION,
+)
+from opentelemetry import trace
+from opentelemetry import metrics
+
+from typing import Iterable
+from flask import Flask, request
+
+
+from opentelemetry.metrics import CallbackOptions, Observation
+
+
+tracer = trace.get_tracer(__name__)
+meter = metrics.get_meter(__name__)
+
+app = Flask(__name__)
+
+cfg = create_config('config.yaml')
+
+"""
+TRACES & REQUEST BASED METRICS
+"""
+# variable to keep track of the total number api requests
+n = 0
+
+# update_total_bytes_sent updates the metric with a value between 0 and 1024
+def update_total_bytes_sent():
+    min = 0 
+    max = 1024
+    total_bytes_sent.add(random.randint(min,max))
+
+# update latency time updates the metric with a value between 0 and 512
+def update_latency_time():
+    min = 0
+    max = 512
+    latency_time.record(random.randint(min, max))
+
+# Starts the callback for totalApiRequests
+def api_requests_callback(options: CallbackOptions):
+    global n
+    n += 1
+    add_api_request = Observation(value=n)
+    print("api_requests called by SDK")
+    yield add_api_request
+
+# Converts otel trace id's to an xray format
+def convert_otel_trace_id_to_xray(otel_trace_id_decimal):
+    otel_trace_id_hex = "{:032x}".format(otel_trace_id_decimal)
+    x_ray_trace_id = TRACE_ID_DELIMITER.join(
+        [
+            TRACE_ID_VERSION,
+            otel_trace_id_hex[:TRACE_ID_FIRST_PART_LENGTH],
+            otel_trace_id_hex[TRACE_ID_FIRST_PART_LENGTH:],
+        ]
+    )
+    return '{{"traceId": "{}"}}'.format(x_ray_trace_id)
+
+# register total bytes sent counter
+total_bytes_sent=meter.create_counter(
+    name="total_bytes_sent",
+    description="Keeps a sum of the total amount of bytes sent while application is alive",
+    unit='mb'
+)
+
+# register api requests observable counter
+total_api_requests=meter.create_observable_counter(
+    name="api_requests",
+    callbacks=[api_requests_callback],
+    description="increments by one every time a sampleapp endpoint is used",
+    unit='1'
+)
+
+# registers latency time histogram
+latency_time=meter.create_histogram(
+    name="latency_time",
+    description="measures latency time",
```

Comment:
```suggestion
    name="latencyTime",
    description="Measures latency time in buckets of 100 300 and 500",
```
Also update the histogram to use buckets of 100 300 and 500